primitives/avn-common/src/lib.rs

```diff
@@ -264,6 +278,38 @@ pub fn hash_with_ethereum_prefix(hex_message: &String) -> Result<[u8; 32], ECDSA
     Ok(hash)
 }
 
+// Be careful when changing this logic because it needs to be compatible with other Ethereum
+// wallets. The message_bytes are converted to hex first then to utf8bytes.
+pub fn hash_string_data_with_ethereum_prefix(
```

Comment:
Would it be possible to use this across the AvN?

Reply:
No, we should use the existing one. This one should be used if we need to validate an externally signed signature.